tests/test_pipe_diameter_sizing.py

```diff
@@ -9,6 +9,23 @@
 
 class TestPipeDiameterSizingExample(TestCase):
     def test_half_network_gone(self):
+        """
+        This test is to check if the optimization behaves as expected under pipe class optimization.
```

Comment:
The problem has no optional sources only optional pipes, I am keeping the original comment

Reply:
You see it correct. The test should be changed, we here hardcode that all pipes are sized via the pipe_classes() method instead of using optional in the ESDL, I add this to the missing in the docstring.